community/index/src/main/java/org/neo4j/index/internal/gbptree/SeekCursor.java

```diff
@@ -1118,12 +1113,7 @@ private boolean verifyNodeGenerationInvariants()
             lastFollowedPointerGeneration = 0;
             expectedCurrentNodeGeneration = currentNodeGeneration;
         }
-        else if ( currentNodeGeneration != expectedCurrentNodeGeneration )
-        {
-            // We've read more than once from this node and between reads the node generation has changed.
-            // This means the node has been reused.
-            return false;
-        }
+        else return currentNodeGeneration == expectedCurrentNodeGeneration;
         return true;
```

Comment:
This can also be simplified to not need the `else` clause. Also, I think we should keep the comment.